core/src/main/scala/fs2/Scheduler.scala

```diff
@@ -0,0 +1,47 @@
+package fs2
+
+import scala.concurrent.duration.FiniteDuration
+import java.util.concurrent.{ Executors, ScheduledExecutorService, TimeUnit }
+
+/** Provides the ability to schedule evaluation of thunks in the future. */
+trait Scheduler {
+
```

Comment:
I was sort of thinking on this earlier. I am not sure if this shouldn't demand `S` in scope, and essentially return to `S` as soon as the timer is fired. Really I think Scheduler shall be only timing service, and never the executor thread, so we are quite sure where the programs run at the end. Any thoughts?